internal/project/project.go

```diff
@@ -561,6 +562,12 @@ func (p *Project) updateProgram() bool {
 			} else {
 				rootFileNames := p.GetRootFileNames()
 				compilerOptions := p.compilerOptions
+
+				if compilerOptions.MaxNodeModuleJsDepth == nil && p.rootJSFileCount > 0 {
+					compilerOptions = compilerOptions.Clone()
```

Comment:
There is method GetCompilerOptions - do you need to modify that instead and scan through other usage of p.compilerOptions ?

Reply:
Tried, but the concurrency story on this is pretty weird. I'd rather just leave this here for now and fix it if we need this option set for something else eventually. It'll probably get easier with the snapshot refactor.